src/bindings/node/node.cpp

```diff
@@ -0,0 +1,125 @@
+#include <node_api.h>
+
+#include <boost/make_shared.hpp>
+#include <boost/noncopyable.hpp>
+#include <boost/optional.hpp>
+#include <boost/property_tree/json_parser.hpp>
+#include <boost/property_tree/ptree.hpp>
+#include <boost/shared_ptr.hpp>
+#include <sstream>
+#include <string>
+#include <iostream>
+
+#include "valhalla/tyr/actor.h"
+
+boost::property_tree::ptree Configuration;
+
+boost::property_tree::ptree json_to_pt(const std::string& json) {
+  std::stringstream ss;
+  ss << json;
+  boost::property_tree::ptree pt;
+  boost::property_tree::read_json(ss, pt);
+  return pt;
+}
+
+boost::property_tree::ptree make_conf(const std::string config) {
+  return json_to_pt(config);
+}
+
+napi_value Route(napi_env env, napi_callback_info info) {
+  napi_status status;
+  napi_value myStr;
+
+  size_t argc = 1;
+  napi_value argv[1];
+
+  status = napi_get_cb_info(env, info, &argc, argv, NULL, NULL);
+
+  if (status != napi_ok) {
+    napi_throw_error(env, NULL, "Failed to parse arguments");
+  }
+
+  char buffer[1024];
+  size_t out;
+
+  status = napi_get_value_string_utf8(env, argv[0], buffer, 1024, &out);
+
+  if (status != napi_ok) {
+    napi_throw_error(env, NULL, "Failed to parse first argument to string");
+  }
+  std::string reqString(buffer, out);
+  std::string route_json;
+  try {
+    valhalla::tyr::actor_t actor(Configuration);
```

Comment:
creating and destroying the `actor` object will be very very unperformant. it would not be useful for a production system that is taking live traffic until the lifecycle of the actor object is the same as the lifecycle of the valhalla node module itself. that design (module-level lifecycle) though isn't really though so this isn't a real suggestion :smile: 

the other option, which allows flexibility and peformance, is to do what the python module does. the valhalla python module can instantiate actor objects whose lifecycle you can control over in python land. i think we should design the node bindings the same way. that is the valhalla node module would have two functions:

- [ ] configure valhalla
- [ ] instantiate an actor object

then yes, the actor object should shadow all the actions/methods in the c++ actor class as they are all pretty much the same as `route` and would be no real extra effort to implement. in other words they are all just json/string in json/string out.